### PR TITLE
Else-if control structure

### PR DESCRIPTION
**Pull request by dirk**
Implementation of #21.

### CODE REVIEW

File: lib/grammar2.pegjs [hunk 1]
```diff
@@ -72,7 +71,13 @@ ctrl = ifctrl
      / forctrl
      / returnctrl
 
-ifctrl     = "if" _ c:innerstmt _ b:block { return p.parseIf(c, b) }
+ifctrl = "if" _ c:innerstmt __ b:block ei:(__ elseifcont)* {
```
Comment (ryngonzalez) [inline lib/grammar2.pegjs]: For the `ei(__ elseifcont)*` does the * indicate that there can be an arbitrary amount of else if expressions? Just trying to learn more about writing PEGs.

File: lib/grammar2.pegjs [hunk 2]
```diff
@@ -72,7 +71,13 @@ ctrl = ifctrl
...
+  ei = ei.map(function (pair) { return pair[1] })
```
Comment (ryngonzalez) [inline lib/grammar2.pegjs]: Why are you transforming the set of ei expression arrays into the second element of the pair?

File: lib/grammar2.pegjs [hunk 3]
```diff
@@ -72,7 +71,13 @@ ctrl = ifctrl
...
+  return p.parseIf(c, b, ei)
+}
+// Continuations of the if control with else-ifs
+elseifcont = "else" __ "if" _ c:innerstmt __ b:block { return p.parseIf(c, b, null) }
```
Comment (ryngonzalez) [inline lib/grammar2.pegjs]: What do `__` and `_` denote?

File: lib/grammar2.pegjs (hunk 1 above)
Comment (dirk) [inline lib/grammar2.pegjs]: Yup! PEG.js uses the the PCRE convention of asterisk for zero-or-more, plus for one-or-more, and question mark for zero or one.

File: lib/grammar2.pegjs (hunk 2 above)
Comment (dirk) [inline lib/grammar2.pegjs]: The `__` gobbles up whitespace, newlines, and any comments. So before that `map` happens `ei` will be an array similar to:

```js
[["\n# Comment\n", AST.If(...)], [" ", AST.If(...)]]
```

The `map` skips over the white-space globs and extracts just those `AST.If` nodes that correspond to each else-if condition.

File: lib/grammar2.pegjs (hunk 3 above)
Comment (dirk) [inline lib/grammar2.pegjs]: `__` is white-space, newlines, and comments. `_` is just white-space (so just space and tab IIRC).

Comment (ryngonzalez) [inline lib/grammar2.pegjs]: Ah, gotcha

File: lib/grammar2.pegjs (hunk 2 above)
Comment (ryngonzalez) [inline lib/grammar2.pegjs]: :+1:

**Pull request merged**